lib/codecs/src/encoding/format/gelf.rs

```diff
@@ -0,0 +1,350 @@
+    }
+
+    /// Determine if input log event is in valid GELF format. Possible return values:
+    ///    - Ok(None)           => The log event is valid GELF
+    ///    - Ok(Some(LogEvent)) => The the log event isn't stricly valid GELF, but was
+    ///                            conformed to be valid.
+    ///    - Err(..)            => The log event is invalid GELF
+    fn validate_event_is_gelf(&mut self, log: &LogEvent) -> vector_core::Result<()> {
+        self.validate_required_fields(log)?;
+        self.validate_field_names_and_values(log)
+    }
+}
+
+impl Default for GelfSerializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Encoder<Event> for GelfSerializer {
+    type Error = vector_core::Error;
+
+    fn encode(&mut self, event: Event, buffer: &mut BytesMut) -> Result<(), Self::Error> {
+        let log = event.as_log();
+        let writer = buffer.writer();
+
+        self.validate_event_is_gelf(log)?;
+
+        // if event was conformed during validation, use it instead
+        if let Some(conformed) = &self.conformed_log {
+            serde_json::to_writer(writer, &conformed)?;
+        } else {
+            serde_json::to_writer(writer, &log)?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::BTreeMap;
+
+    use super::*;
+    use value::Value;
+    use vector_common::btreemap;
+    use vector_core::event::{Event, EventMetadata};
+
+    fn do_serialize(
+        expect_success: bool,
+        event_fields: BTreeMap<String, Value>,
+    ) -> Option<serde_json::Value> {
+        let config = GelfSerializerConfig::new();
+        let mut serializer = config.build();
+        let event: Event = LogEvent::from_map(event_fields, EventMetadata::default()).into();
+        let mut buffer = BytesMut::new();
+
+        if expect_success {
+            assert!(serializer.encode(event, &mut buffer).is_ok());
+            let buffer_str = std::str::from_utf8(&buffer).unwrap();
+            let result = serde_json::from_str(buffer_str);
+            assert!(result.is_ok());
+            Some(result.unwrap())
+        } else {
+            assert!(serializer.encode(event, &mut buffer).is_err());
+            None
+        }
+    }
+
+    #[test]
+    fn gelf_serializing_valid() {
+        let event_fields = btreemap! {
+            VERSION => "1.1",
+            HOST => "example.org",
+            SHORT_MESSAGE => "Some message",
+            FULL_MESSAGE => "Even more message",
+            FACILITY => "",
+            FILE => "/tmp/foobar",
+            LINE => Value::Float(ordered_float::NotNan::new(1.5).unwrap()),
+            LEVEL => 5,
+        };
+
+        let jsn = do_serialize(true, event_fields).unwrap();
+
+        assert_eq!(jsn.get(VERSION).unwrap(), "1.1");
+        assert_eq!(jsn.get(HOST).unwrap(), "example.org");
+        assert_eq!(jsn.get(SHORT_MESSAGE).unwrap(), "Some message");
+    }
+
+    #[test]
+    fn gelf_serializing_invalid_sanitize() {
```

Comment:
I don't _love_ the name, I think we should unify on sanitize or conform (or something else) and then the function could be `gelf_serializing_conformable`

Reply:
yes, good call. that name is remnant from the eradicated config option `sanitize`